### ISSUE TITLE
regression v8.2.2: optional chained properties reported as undefined

### ISSUE THREAD

**Issue by dnalborczyk** [babel/babel-eslint#595]
it appears that v8.2.2 is reporting optional chained properties as undefined (works with v8.2.1)

```javascript
foo?.bar?.foobar
```

```console
77:15  error  'bar' is not defined     no-undef
77:22  error  'foobar' is not defined  no-undef
```

**Comment by Rachomir** [comment]
I approached same issue

**Comment by lordgiotto** [comment]
I'm facing the same issue.
Here my .babelrc.js

```js
const isProd = process.env.NODE_ENV === 'production';

module.exports = {
  presets: [
    'react-app'
  ],
  plugins: [
    '@babel/plugin-proposal-optional-chaining',
    isProd && 'transform-react-remove-prop-types'
  ].filter(Boolean)
}
```

Thanks for your great work ;)

**Comment by lehni** [comment]
I am seeing this too. Downgrading back to `8.2.1` resolves the issue for me.

**Comment by existentialism** [comment]
Ref: https://github.com/babel/babel/pull/7288

We need to now handle `OptionalMemberExpressions` and `OptionalCallExpressions`.

Added this to my TODO list if no one gets to it first!

**Comment by julien-f** [comment]
@existentialism babel/babel#7288 has been resolved, does this mean this can be fixed?

**Comment by lehni** [comment]
Not a surprise given the state of discussions, but just to prevent other from having to test it: `8.2.3` still suffers from this issue.

**Comment by conradoramalho** [comment]
I downgrade `8.2.1` and this doesn't solves the problem for me. How you fix it?

**Comment by brneto** [comment]
I'm using:
```
ESLint: 4.19.1
Babel: 7.0.0-beta.46
babel-eslint: 8.2.3
@babel/plugin-proposal-optional-chaining: 7.0.0-beta.46
```
And I having the same issue [here](https://github.com/babel/babel-eslint/issues/595#issue-298934862).

Have someone found how to solver that?

**Comment by gastonmorixe** [comment]
hours on this... downgrading to 8.2.1 works for me. Thanks

**Comment by hansonw** [comment]
It seems like this should be fixed once the babylon dep (`@babel/parser` now) is upgraded past beta45 (which includes https://github.com/babel/babel/pull/7668). Right now it's at beta44.

**Comment by lehni** [comment]
@hansonw that alone doesn't seem to fix it. I just tried by installing and linking `babel-eslint` locally, and upgrading its dependencies. I still get the same linting errors with its `@babel` dependencies updated to `beta.48`

**Comment by brneto** [comment]
In fact, downgrading to 8.2.1 have worked. Thanks!

**Comment by richardgirges** [comment]
I'm seeing it with `beta.49`. I too had to downgrade to `8.2.1`

**Comment by brneto** [comment]
@richardgirges Did you see with `beta.49`? Did Work?

**Comment by LennyPenny** [comment]
Even downgrading to `8.2.1` doesn't fix this for me

**Comment by brneto** [comment]
We need the see your .babelrc and .eslintrc, at least.

**Comment by psybork** [comment]
Downgraded to 8.2.1 and I got rid of the 'undefined' errors. Now it seems to be expecting a ternary (probably due to the '?').

Here are my dependencies
```
    "@babel/core": "^7.0.0-beta.49",
    "@babel/plugin-proposal-optional-chaining": "^7.0.0-beta.49",
    "@babel/plugin-syntax-optional-chaining": "^7.0.0-beta.49",
    "@babel/polyfill": "^7.0.0-beta.49",
    "@babel/preset-env": "^7.0.0-beta.49",
    "@babel/preset-react": "^7.0.0-beta.49",
    "babel-eslint": "^8.2.1",
    "babel-loader": "^8.0.0-beta.3",
    "eslint": "^4.19.1",
```
My .babelrc and .eslintrc, respectively

```
{
  "presets": [
    "@babel/react",
    "@babel/preset-env"
  ],
  "plugins": [
    "@babel/plugin-proposal-optional-chaining",
    "react-hot-loader/babel"
  ]
}
```

```
{
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": ["eslint:recommended", "airbnb"],
  "parserOptions": {
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true,
      "jsx": true
    },
    "sourceType": "module"
  },
  "plugins": ["react", "jsx-a11y", "import"],
  "rules": { ... }
```

**Comment by LennyPenny** [comment]
@brneto 
`.babelrc`
```json
{
    "presets": [
        [
            "@babel/preset-env",
            {
                "targets": {
                    "node": "current"
                }
            }
        ]
    ],
    "sourceMaps": "inline",
    "plugins": [
        "@babel/plugin-proposal-optional-chaining",
        "import-glob"
    ]
}
```

`.eslintrc`
```json
{
    "parser": "babel-eslint",
    "env": {
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 8,
        "sourceType": "module"
    },
    "rules": {...}
}
```

**Comment by brneto** [comment]
@LennyPenny Are you importing `@babel/polyfill` at the root of your code?

**Comment by rubennorte** [comment]
@existentialism I guess I got to it first. Can you please take a look at #630 where I put a fix for this? This is the last thing we need in order to roll out this at Facebook. Thanks!

**Comment by LennyPenny** [comment]
@brneto I'm using node with the latest babel. I never needed `@babel/polyfill` before, but even with it it doesn't work

**Comment by lehni** [comment]
Unfortunately we're not out of the woods yet: #643

**Comment by jlaustill** [comment]
@lehni What is the process of helping with this?  I'd like to get to a point that I can use optional chaining with eslint and cra@next, but I'm not sure how to plug in to help.

**Comment by fnlctrl** [comment]
This should probably be reopened as 8.2.6 still doesn't work

**Comment by macrozone** [comment]
even with assigments this error gets thrown using babel-eslint 9.0.0 final

https://github.com/babel/babel-eslint/issues/643#issuecomment-417282457

downgrading to 8.2.1 solves it.
